src/main/java/com/thealgorithms/datastructures/trees/SplayTree.java

```diff
@@ -0,0 +1,323 @@
+                root = rotateLeft(root);
+            }
+
+            return (root.right == null) ? root : rotateLeft(root);
+        }
+    }
+
+    /**
+     * Insert a key into the SplayTree.
+     *
+     * @param key The key to insert.
+     */
+    public void insert(int key) {
+        root = insertRec(root, key);
+        root = splay(root, key);
+    }
+
+    /**
+     * Recursive function to insert a key.
+     *
+     * @param root The root of the subtree to insert the key into.
+     * @param key  The key to insert.
+     * @return The root of the modified subtree.
+     */
+    private Node insertRec(Node root, int key) {
+        if (root == null) return new Node(key);
+
+        if (root.key > key) {
+            root.left = insertRec(root.left, key);
+        } else if (root.key < key) {
+            root.right = insertRec(root.right, key);
+        }
+
+        return root;
+    }
+
+    /**
+     * Search for a key in the SplayTree.
+     *
+     * @param key The key to search for.
+     * @return True if the key is found, otherwise false.
+     */
+    public boolean search(int key) {
+        root = splay(root, key);
+        return root != null && root.key == key;
+    }
+
+    /**
+     * Delete a key from the SplayTree.
+     *
+     * @param key The key to delete.
+     */
+    public void delete(int key) {
+        root = deleteRec(root, key);
+    }
+
+    /**
+     * Recursive function to delete a key.
+     *
+     * @param root The root of the subtree to delete the key from.
+     * @param key  The key to delete.
+     * @return The root of the modified subtree.
+     */
+    private Node deleteRec(Node root, int key) {
+        if (root == null) return null;
+
+        if (root.key > key) {
+            root.left = deleteRec(root.left, key);
+        } else if (root.key < key) {
+            root.right = deleteRec(root.right, key);
+        } else {
+            // Found the node to delete
+            if (root.left == null)
+                return root.right;
+            else if (root.right == null)
+                return root.left;
+
+            // Node with two children: Get the inorder successor (smallest in the right subtree)
+            root.key = minValue(root.right);
+
+            // Delete the inorder successor
+            root.right = deleteRec(root.right, root.key);
+        }
+
+        return root;
+    }
+
+    /**
+     * Find the minimum value in a subtree.
+     *
+     * @param root The root of the subtree to find the minimum value in.
+     * @return The minimum value in the subtree.
+     */
+    private int minValue(Node root) {
+        int minValue = root.key;
+        while (root.left != null) {
+            minValue = root.left.key;
+            root = root.left;
+        }
+        return minValue;
+    }
+
+    /**
+     * Perform an in-order traversal of the SplayTree.
+     *
+     * @return A vector containing the keys in in-order traversal order.
```

Comment:
vector or list?

Reply:
I am confused, I'll update the docstrings